_posts/2025-02-18-agentic-ai-with-quarkus.adoc

```diff
@@ -0,0 +1,178 @@
+3. a https://github.com/mariofusco/quarkus-agentic-ai/blob/main/src/main/java/org/agenticai/aiastool/weather/WeatherForecastService.java[second web service] providing the weather forecast for the given latitude and longitude.
+
+.Weather agent architecture
+image::weather-agent.png[]
+
+It's possible to put at work the https://github.com/mariofusco/quarkus-agentic-ai/blob/main/src/main/java/org/agenticai/aiastool/WeatherResource.java[rest endpoint] exposing this agent-based weather service, for example asking it what's the weather in Rome
+
+http://localhost:8080/weather/city/Rome
+
+and receive an answer like the following.
+
+----
+The weather in Rome today will have a maximum temperature of 14.3°C, minimum temperature of 2.0°C.
+No precipitation expected, and the wind speed will be up to 5.6 km/h.
+The overall weather condition is expected to be cloudy.
+----
+
+In essence this control flow is quite similar to what described in the prompt chaining workflow, where the user input is sequentially transformed in steps (in this case going from the prompt, to the name of the city contained in that prompt, to the geographical coordinates of that city, to the weather forecasts at those coordinates), but as anticipated this time the major difference is in the fact that the sequence of steps is directly scheduled by the LLM instead of being externally orchestrated in a programmatic way.
+
+=== A more general purpose AI agent
+
+Using a bigger model (https://github.com/mariofusco/quarkus-agentic-ai/blob/main/src/main/java/org/agenticai/searchastool[this example] has been configured with `qwen2.5-14b` and a longer timeout to give it a chance to complete its task)
+
+[source, properties]
+----
+quarkus.langchain4j.ollama.big-model.chat-model.model-id=qwen2.5:14b
+quarkus.langchain4j.ollama.big-model.chat-model.temperature=0
+quarkus.langchain4j.ollama.big-model.timeout=600s
+----
+
+it is possible to dare to implement an AI agent with more freedom to operate, in order to make it capable to cover a larger range of tasks. This https://github.com/mariofusco/quarkus-agentic-ai/blob/main/src/main/java/org/agenticai/searchastool/IntelligentAgent.java[intelligent agent] has been provided with only https://github.com/mariofusco/quarkus-agentic-ai/blob/main/src/main/java/org/agenticai/searchastool/WebSearchService.java[one tool] that can perform a web search on DuckDuckGo and return the result in plain text. What the AI agent does is using this tool to retrieve online information for things that it ignores and put them together in order to provide an answer for a generic user question.
+
+For instance we may want to ask `how many seconds would it take for a leopard at full speed to run through Pont des Arts` making a request to the https://github.com/mariofusco/quarkus-agentic-ai/blob/main/src/main/java/org/agenticai/searchastool/AgenticChatbotResource.java[rest endpoint] exposing this service like it follows
```

Comment:
```suggestion
For instance, consider the following question: How many seconds would it take for a leopard at full speed to run through the Pont des Arts? Using this https://github.com/mariofusco/quarkus-agentic-ai/blob/main/src/main/java/org/agenticai/searchastool/AgenticChatbotResource.java[HTTP endpoint], it would be executed using:
```